homeassistant/components/becker/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""The becker component."""
+import logging
+
+from pybecker.becker import Becker
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT
+
+from .const import (
+    CONF_CHANNEL,
+    CONF_COVERS,
+    CONF_DEVICE,
+    CONF_UNIT,
+    DEFAULT_CONF_USB_STICK_PATH,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PAIR_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CHANNEL): vol.All(int, vol.Range(min=1, max=7)),
+        vol.Optional(CONF_UNIT): vol.All(int, vol.Range(min=1, max=5)),
+    }
+)
+
+
+async def async_setup_entry(hass, entry):
+    """Establish connection with Becker Centronic."""
+    conf = entry.data
+
+    stick_path = conf[CONF_DEVICE]
+    # if stick_path is not set in integration try to get it from configuration
+    if not stick_path:
+        stick_path = hass.data[DOMAIN][CONF_DEVICE]
+
+    _LOGGER.debug("SETTING CENTRONIC STICK ON PORT {stick_path}")
+    becker = BeckerConnection(stick_path)
+
+    if not becker:
+        return False
+
+    for _ in range(0, 2):
+        _LOGGER.debug("Init call to cover channel 1")
+        await becker.connection.stop("1")
+
+    hass.data[DOMAIN]["connector"] = becker.connection
+
+    hass.data.setdefault(DOMAIN, {}).update({entry.entry_id: becker})
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "cover")
+    )
+
+    hass.services.async_register(DOMAIN, "pair", becker.handle_pair, PAIR_SCHEMA)
+    hass.services.async_register(DOMAIN, "log_units", becker.handle_log_units)
+    return True
+
+
+async def async_setup(hass, config):
+    """Set up the Becker platform."""
+    conf = config.get(DOMAIN)
+    if conf is None:
+        conf = {}
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][CONF_COVERS] = {}
+
+    # User has configured covers
+    if CONF_COVERS not in conf:
+        return True
+    covers = conf[CONF_COVERS]
+
+    if CONF_DEVICE in conf:
+        hass.data[DOMAIN][CONF_DEVICE] = conf[CONF_DEVICE]
+
+    for cover_conf in covers:
+        channel = cover_conf[CONF_CHANNEL]
+        # Store config in hass.data so the config entry can find it
+        hass.data[DOMAIN][CONF_COVERS][channel] = cover_conf
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data={"cover": cover_conf}
```

Comment:
This is a new integration, there is nothing to import from previous integrations? I guess this can be removed?

Reply:
Hi @frenck thank you for your feedback, I've introduced unit tests and removed the french translation.

Reply:
This piece of code is to retrieve the covers from the configuration file (they need to be manually setup) the config flow is setting up the interface (usually the rf usb stick from becker)

Reply:
Ask the user to configure devices during the config flow or use an options flow to get this data from the users.

Reply:
I also see that there is a service too. Shouldn't that be enough ?

Reply:
@nicolasberthel could you answer this question please.

Reply:
@balloob  the service is "just" for the pairing (to pair the cover to the usb stick), the cover setup is not yet done via the UI, basically this is the list of covers that is still in the yaml file (i.e. documentation)